### ISSUE TITLE
fix or error-trap that model variable names can't start with "logProb"

### ISSUE THREAD

**Issue by paciorek** [nimble-dev/nimble#1210]
Here's a (non-minimal) reprex. The problem is that the dummy variable starts with "logProb", which presumably causes some issue with internal processing expecting only logProb quantities to have their name start with "logProb". This causes NA values in mvSamples in uncompiled running and a compilation error in compilation of the MCMC.

```
## sampler to monitor logProbabilities of all nodes in the model
## or only of a set of nodes if provided

logProb_summer <- nimbleFunction(
    name = 'logProb_summer',
    contains = sampler_BASE,
    setup = function(model, mvSaved, target,  control) {

        if(!is.null(control$nodeList)){
            nodes <- model$expandNodeNames(control$nodeList)
        } else {
        ## use all nodes in the model if the user does not provide
        ## a list of nodes to be monitored
            nodes <- model$getNodeNames()
        }
        ## esclude target dummy node
        nodes <- nodes[nodes != target]

    },
    run = function() {
        model[[target]] <<- model$getLogProb(nodes)  
        
        copy(from = model, to = mvSaved, 
             row = 1, nodes = target, logProb = FALSE)
    }, 
    methods = list( reset = function () {} )
)
 
code <- nimbleCode({
  for(i in 1:N) {
    y[i] ~ dnorm(mu, var = 1)
  }
  mu ~ dnorm(0,1)
  
  ## dummy node to track log probs
  logProbAll  ~ dnorm(0,1)

})

data <- list(y = rnorm(100))
constants <- list(N = length(data$y))

inits <- list(mu = 0)

model <- nimbleModel(code  = code,
                    data            = data,  
                    constants   = constants,
                    inits           = inits, 
                    calculate   = FALSE)

monitors <- c("mu", "logProbAll")

mcmcConf <- configureMCMC(model, monitors = monitors)
mcmcConf$removeSampler("logProbAll")
mcmcConf$addSampler("logProbAll", type =  "logProb_summer")

mcmcConf

mcmc <- buildMCMC(mcmcConf) 

Cmodel <- compileNimble(model)

## SP: Error when compiling the MCMC
# Error in eval(call(".Call", nimbleUserNamespace$sessionSpecificDll$populateValueMapAccessorsFromNodeNames,  :
#   VECTOR_ELT() can only be applied to a 'list', not a 'NULL'

Cmcmc <- compileNimble(mcmc, project = model, 
  showCompilerOutput = TRUE)


res <- runMCMC(Cmcmc, 
           niter    = 1000,  
           nburnin  = 100, 
           setSeed  = 32)
```

**Comment by paciorek** [comment]
Here's a simpler reprex:

```
code <- nimbleCode({
    y ~ dnorm(mu,1)
    mu ~ dnorm(0,1)
    logProb ~ dnorm(0,1)
})
m <- nimbleModel(code,data=list(y=3))
cm <- compileNimble(m)
mcmc <- buildMCMC(m)
cmcmc <- compileNimble(mcmc,project=m)
```
```
Error in eval(call(".Call", nimbleUserNamespace$sessionSpecificDll$populateValueMapAccessorsFromNodeNames,  : 
  VECTOR_ELT() can only be applied to a 'list', not a 'NULL'
```

**Comment by paciorek** [comment]
@danielturek I think this issue arises because in line 350 of `MCMC_utils.R` we check for logProb names in `nodes`:

```
    isLogProbName <- grepl('logProb', nodes)
```

so any valid node name that has 'logProb' in it causes downstream problems.

I'd like to change the regex pattern to be '^logProb_' there and in the `gsub`, but I wanted to run this by you in case anything else occurs to you.

**Comment by danielturek** [comment]
@paciorek LGTM.

**Comment by paciorek** [comment]
Fixed by #1240
